sycl/unittests/Extensions/CommandGraph/GraphFusion.cpp

```diff
@@ -0,0 +1,113 @@
+//==------------------------- Exceptions.cpp -------------------------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "Common.hpp"
+
+using namespace sycl;
+using namespace sycl::ext::oneapi;
+
+#if SYCL_EXT_CODEPLAY_KERNEL_FUSION
+
+// Various tests which are checking for correct exception behaviour
+// for Graph Fusion.
```

Comment:
could also use a test that if updatable property is passed in finalize an error is thrown